### ISSUE TITLE
Locale null countryCode in flutter internationalization

### ISSUE THREAD

**Issue by pq** [flutter/flutter#16667]
_From @MarkBriggs2 on April 17, 2018 14:40_

Question about  #internationalization
Working through the tutorial at https://flutter.io/tutorials/internationalization/<br>
Source at: 
https://github.com/flutter/website/tree/master/_includes/code/internationalization/intl/<br>
If supportedLocales parameter to `MaterialApp` constructor is as follows:
```dart
supportedLocales: [
    const Locale('en', 'US'), 
    const Locale('es', 'ES'), 
  ],
```
this initially works. If country codes are omitted: 
```dart
supportedLocales: [
    const Locale('en'), 
    const Locale('es'), 
  ],
```
 then the `countryCode` getter of the Locale object passed to `DemoLocalizations.load()` or `GlobalMaterializations._computeLocaleName(Locale locale)` (`material_localizations.dart` line 134) will be null so any attempt to access `locale.countryCode.isEmpty` will generate a `NoSuchMethodError`.  
Is this as intended?  
Thanks

_Copied from original issue: flutter/flutter-intellij#2093_

**Comment by MarkBriggs2** [comment]
Locale constructor is:
<pre>
const Locale(this._languageCode, [ this._countryCode ]) : assert(_languageCode != null);
</pre>
If altering member <code>_countyCode<code> to default to empty string is not recommended, then perhaps the solution may be to alter  <code>GlobalMaterializations._computeLocaleName(Locale locale) </code> (from material_localizations.dart) to check for null. Then modify the tutorial so that the app specific localization instance's <code>load()</code> does the same. 
Just a suggestion...:)

**Comment by dooboolab** [comment]
I am facing below error in real device. don't know why.
```
Xcode build done.
Installing and launching...
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/usr/local/Cellar/python@2/2.7.14_3/Frameworks/Python.framework/Versions/2.7/lib/python2.7/copy.py", line 52, in <module>
    import weakref
  File "/usr/local/Cellar/python@2/2.7.14_3/Frameworks/Python.framework/Versions/2.7/lib/python2.7/weakref.py", line 14, in <module>
    from _weakref import (
ImportError: cannot import name _remove_dead_weakref
flutter: locale: en_US
flutter: locale: ko_KR
Syncing files to device 장효찬의 iPhone...
flutter: Load en
[VERBOSE-2:dart_error.cc(16)] Unhandled exception:

```

**Comment by cbracken** [comment]
@dooboolab do you mean the Python error? If so, I believe that should be fixed via #19281. It should make it to the next beta release. You can verify that on dev channel via `flutter channel dev` (`flutter channel beta` to switch back to beta channel).

If you mean the unhandled exception below that, it looks like the exception and any stack trace were truncated from your post. If you're still seeing this, would you mind opening a new bug with a verbose log (`flutter --verbose run ...`)?
